### ISSUE TITLE
Promises/A+ Interface

### ISSUE THREAD

**Issue by LPGhatguy** [SBoudrias/Inquirer.js#268]
With Promise being implemented in io.js, and being merged into mainstream Node as of ~4.0, it might be a good idea to provide a native Promise implementation.

In my code right now, I wrap Inquirer:

```js
inquirer.prompt(questions)
	.then((answers) => {
		console.log("ANSWERS!");
	});
```

**Comment by SBoudrias** [comment]
Once it'll be available in native node, then we'll make it happen.

But this is not yet the case.

**Comment by sindresorhus** [comment]
Promises are already available in Node.js 0.12, and if you're not willing to drop Node.js 0.10 support, you can use a tiny polyfill like [`pinkie-promise`](https://github.com/floatdrop/pinkie-promise).

**Comment by LPGhatguy** [comment]
@SBoudrias Node v4 landed today which has nearly full ES6 support now. Node 0.12 had promises as well.

Native node has landed.

**Comment by TrySound** [comment]
:+1:

**Comment by TrySound** [comment]
Also it would be good to return promise in async callbacks instead of `this.async();`

**Comment by pgherveou** [comment]
That would be great especially when used with generator function 
```js
co(function*() {
  const answers = yield inquirer.prompt(questions)
})
```

**Comment by callumlocke** [comment]
@SBoudrias I am happy to put together a PR if you can confirm you're willing to do a major version bump for this?

**Comment by SBoudrias** [comment]
@callumlocke PR welcomed

**Comment by AGresvig** [comment]
@SBoudrias did you ever get around to it? I agree Promises is the way to go now, callbacks are a thing of the past in my workspace.

**Comment by SBoudrias** [comment]
PR welcomed as it's now officially supported since Node v4

**Comment by callumlocke** [comment]
@AGresvig I had a dig around but decided I'm not the one to do it if @SBoudrias wants to retain the Rx stuff.

**Comment by SBoudrias** [comment]
We can return a promise and add the rx property on it.

```js
var promise = new Promise(/* etc */);
promise.process = rx.js.stuff;
return promise;
```

**Comment by Dianoga** [comment]
Is this change stable enough to release a new version?

**Comment by SBoudrias** [comment]
Soon, it'll be released with 1.0 you can see the issues left on the milestone here: https://github.com/SBoudrias/Inquirer.js/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0.0

Help is welcomed to make it happen faster.
